File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int result = 1;
        HashMap<String, Integer> clothesMap = new HashMap<>();
        
        for (String[] cloth : clothes) {
            clothesMap.put(cloth[1], clothesMap.getOrDefault(cloth[1], 0) + 1);
        }
        for (int c : clothesMap.values()) {
            result *= c + 1;
        }
        return result - 1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    dic = dict()
    sum = 1
    for i in range(len(clothes)):
        if clothes[i][1] in dic:
            dic[clothes[i][1]] += 1
        else:
            dic[clothes[i][1]] = 1
    
    for _,su in dic.items():
        sum *= su+1
        
    return sum-1